feature/google/src/test/java/com/revenuecat/purchases/google/FindBestOfferTest.kt

```diff
@@ -0,0 +1,529 @@
+         *   |-> monthly (P1M, $4.99) (BEST OFFER)
+         *         |
+         *         |-> month-free [rc-ignore-best-offer]
+         *               |-> (P1M, free)
+         */
+
+        // Base plan
+        var monthlyBasePlanPricingPhase = mockPricingPhase(
+            price = 4.99,
+            billingPeriod = "P1M"
+        )
+        val monthlyBasePlan = mockSubscriptionOfferDetails(
+            offerId = "",
+            basePlanId = "monthly",
+            pricingPhases = listOf(monthlyBasePlanPricingPhase)
+        )
+
+        // month-free (P1M, free)
+        val monthFreeOffer = mockPricingPhase(
+            price = 0.0,
+            billingPeriod = "P1M",
+            billingCycleCount = 1,
+            recurrenceMode = ProductDetails.RecurrenceMode.NON_RECURRING
+        )
+        val freeOffer = mockSubscriptionOfferDetails(
+            tags = listOf("rc-ignore-best-offer"),
+            offerId = "month-free",
+            basePlanId = "monthly",
+            pricingPhases = listOf(monthFreeOffer, monthlyBasePlanPricingPhase)
+        )
+
+        // Product details
+        val productDetail1 = mockProductDetails(
+            productId = "sub_1",
+            type = BillingClient.ProductType.SUBS,
+            subscriptionOfferDetails = listOf(monthlyBasePlan, freeOffer))
+        val productDetails = listOf(productDetail1)
+
+        val storeProducts = productDetails.toStoreProducts()
+
+        val defaultOption = storeProducts.first().defaultOption as GoogleSubscriptionOption
+        assertThat(defaultOption.id).isEqualTo("monthly")
+    }
+
+    @Test
+    fun `default offer is offer with longest free phase without rc-ignore-best-offer tag`() {
+        /*
+         * sub_1
+         *   |
+         *   |-> monthly (P1M, $4.99)
+         *   |     |
+         *   |     |-> week-free
+         *   |     |     |-> (P1W, free)
+         *   |     |-> two-week-free (BEST OFFER)
+         *   |     |     |-> (P2W, free)
+         *   |     |-> monthly-free [rc-ignore-best-offer]
+         *   |     |     |-> (P1M, free)
+         *   |     |-> nine-day-free
+         *   |           |-> (P9D, free
+         */
+
+        // Base plan
+        var monthlyBasePlanPricingPhase = mockPricingPhase(
+            price = 4.99,
+            billingPeriod = "P1M"
+        )
+        val monthlyBasePlan = mockSubscriptionOfferDetails(
+            offerId = "",
+            basePlanId = "monthly",
+            pricingPhases = listOf(monthlyBasePlanPricingPhase)
+        )
+
+        // week-free (P1M, free)
+        val weekFreeOffer = mockSubscriptionOfferDetails(
+            offerId = "week-free",
+            basePlanId = "monthly",
+            pricingPhases = listOf(mockPricingPhase(
+                price = 0.0,
+                billingPeriod = "P1W",
+                billingCycleCount = 1,
+                recurrenceMode = ProductDetails.RecurrenceMode.NON_RECURRING
+            )) + monthlyBasePlanPricingPhase
```

Comment:
Minor: i'd make a few helper function for these, like `freePhase("P1W")` to make the tests easier to read.